crates/bevy_ecs/src/schedule/runner/mod.rs

```diff
@@ -0,0 +1,190 @@
+}
+
+impl Default for Runner {
+    fn default() -> Self {
+        Self {
+            systems_topsort: Vec::new(),
+            systems: Vec::new(),
+            system_conditions: Vec::new(),
+            system_deps: HashMap::new(),
+            system_sets: HashMap::new(),
+            sets_topsort: Vec::new(),
+            set_conditions: Vec::new(),
+            set_systems: HashMap::new(),
+        }
+    }
+}
+
+impl Runner {
+    pub(crate) fn simple(self) -> SimpleRunner {
+        SimpleRunner::with(self)
+    }
+
+    pub(crate) fn single_threaded(self) -> SingleThreadedRunner {
+        SingleThreadedRunner::with(self)
+    }
+}
+
+/// Runs the system or system set given by `schedule` on the world.
+pub fn run_systems(schedule: impl SystemLabel, world: &mut World) {
+    let mut reg = world.resource_mut::<SystemRegistry>();
+    let id = *reg.ids.get(&schedule.dyn_clone()).expect("unknown label");
+    match id {
+        RegistryId::System(_) => {
+            let mut reg = world.resource_mut::<SystemRegistry>();
+            let mut system = reg.systems.get_mut(&id).unwrap().take().unwrap();
+            let mut system_conditions = reg.system_conditions.get_mut(&id).unwrap().take().unwrap();
+
+            let mut should_run = true;
+            for cond in system_conditions.iter_mut() {
+                should_run &= cond.run((), world);
+            }
+
+            if should_run {
+                system.run((), world);
+                system.apply_buffers(world);
+            }
+
+            let mut reg = world.resource_mut::<SystemRegistry>();
+            reg.systems
+                .get_mut(&id)
+                .map(|container| container.insert(system));
+
+            reg.system_conditions
+                .get_mut(&id)
+                .map(|container| container.insert(system_conditions));
+        }
+        RegistryId::Set(_) => {
+            let mut reg = world.resource_mut::<SystemRegistry>();
+            let mut runner = reg.runners.get_mut(&id).unwrap().take().unwrap();
+            assert!(runner.systems.is_empty());
+            assert!(runner.system_conditions.is_empty());
+            assert!(runner.set_conditions.is_empty());
+
+            for sys_id in runner.systems_topsort.iter() {
+                runner
+                    .systems
+                    .push(reg.systems.get_mut(&sys_id).unwrap().take().unwrap());
+
+                runner.system_conditions.push(
+                    reg.system_conditions
+                        .get_mut(&sys_id)
+                        .unwrap()
+                        .take()
+                        .unwrap(),
+                );
+            }
+
+            for set_id in runner.sets_topsort.iter() {
+                runner
+                    .set_conditions
+                    .push(reg.set_conditions.get_mut(&set_id).unwrap().take().unwrap());
+            }
+
+            let mut runner = runner.single_threaded();
```

Comment:
Ah: you're special-casing this for now?

Reply:
Lol yeah, multi-threaded executor is blocked on the nested task scope issue.